sips/sip-60.md

```diff
@@ -54,24 +55,68 @@ TBD: Should the escrowed SNX be liquidated at a larger discount? or vested and t
+    function accountMigratedToL2(address account) external view returns (bool);
+
     // Updated to handle paging of 100 escrow entries at a time from a starting vesting index
-    function checkAccountSchedule(address account, uint startIndex) public view returns (uint[100] memory) 
+    function checkAccountSchedule(address account, uint startIndex) public view returns (uint[100] memory)
 
     // Mutative functions
-    
+
     // Added escrowLength argument allowing any length of escrow period from 1 second to x years
     function appendVestingEntry(address account, uint quantity, uint escrowLength) external;
-        
+
     function vest() external;
-    
+
     // New function to allow any account vest another account
     function vestOnBehalf(address account) external;
 }
 ```
+
 ### Migration
-1. Synthetix will need to be upgraded to migrate the SNX to the new contract onchain.
-2. During the migration `Vest` needs to be disabled or effectivly fail to ensure integrity of escrow entries and SNX balances beign migrated for all accounts.
+
+1. Issuer contract will need to be upgraded to migrate the SNX to the new contract onchain.
+
+2. During the migration `Vest` needs to be disabled or effectivly fail to ensure integrity of escrow entries and SNX balances being migrated for all accounts.
+
+3. `totalEscrowedAccountBalance` will be migrated by the protocol for all accounts on the OLD reward escrow to ensure users are allocated their collateral for staking on the new reward escrow contract.
+
+4. Claiming rewards and vesting on the new contract will be blocked for users with existing vesting entries.
+
+5. Users with existing vesting entries on the old rewards escrow will need to migrate their vesting entries across to the new contract. The migration will read from the address's existing entries and copy them across.
+
+6. Any escrow SNX that can be vested will be vested first. This will mean that only the remaining 52 weeks of vesting entries will be copied.
+
+### L2 Escrow Migration
+
+With the launch of L2 Staking for SNX on the OVM testnet, users will be able to migrate all their SNX and escrowed SNX to L2 for staking and rewards. The vesting entries will be copied onto the L2 reward escrow contract that mirrors the migration process.
+
+1. User first approves on the Reward Escrow for the `SecondaryDeposit` contract to transfer their escrowed SNX when `secondaryDeposit.deposit()` is called. The approval transaction will also vest the SNX so that the user will only migrate the remaining 52 weeks of vesting entries to L2.
+
+2. The L1 migration step is not required for stakers to migrate to L2 their escrowed SNX. If the user has not migrated on L1, the `secondaryDeposit.deposit()` function will read from the Old RewardEscrow to determine the remaining escrowed SNX and entries to be migrated to L2. This reduces the steps and costs for stakers who want to move to L2 so that they won't need to pay the gas costs of L1 migration.
```

Comment:
This isn't what was discussed and it's very problematic. It means that there's SNX on L1 that could be vested which is also on L2.

Reply:
Should be resolved if we block migration from the old to new escrow on L1 when a user has already migrated to L2.